File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hmap = {}
        for i in range(len(nums)):
            cval = target - nums[i]
            if cval in hmap:
                return [i, hmap[cval]]
            hmap[nums[i]] = i
        

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let cache = {}
    let cval
    for (let i = 0; i < nums.length; i++) {
        cval = target - nums[i]
        if (cval in cache) return [i, cache[cval]]
        cache[nums[i]] = i
    }
    return
};